awscli/testutils.py

```diff
@@ -789,7 +792,13 @@ def put_object(self, bucket_name, key_name, contents='', extra_args=None):
     def delete_bucket(self, bucket_name):
         self.remove_all_objects(bucket_name)
         client = self.create_client_for_bucket(bucket_name)
-        response = client.delete_bucket(Bucket=bucket_name)
+
+        # Ironically there's a chance that, even though the bucket has been
+        # used several times, the delete will fail due to eventual
+        # consistency issues. So we need to make sure the bucket actually
+        # exists before deleting it.
+        self.wait_bucket_exists(bucket_name, min_successes=5)
+        client.delete_bucket(Bucket=bucket_name)
```

Comment:
This still isn't guaranteed to work, the issue is that despite other operations succeeding (List and potentially even HeadBucket), the *delete_bucket* call fails.  What about changing this slightly to minimize the chances of this failing:

1. Try to delete the bucket
2. Catch the NoSuchBucket error
3. If the bucket doesn't exist, then we're good.
4. Otherwise retry up to N times.

Reply:
For 3) are you suggesting to do a head_bucket to determine if it exists?